checkers/ifreturnslit.ml

```diff
@@ -0,0 +1,27 @@
+open Asttypes
+open Parsetree
+open Longident
+open Check
+open Report
+
+module IfReturnsLit : CHECK = struct
+
+  let check (exp : expr) : warn option =
+    begin match exp with
+    | EIfThenElse (test, b_then, b_else) ->
+      begin match b_then, b_else with
+        | Pexp_construct ({txt = Lident "true"}, _),
+          Pexp_construct ({txt = Lident "false"}, _) ->
+          Some ({
+              loc = (warn_loc 1 1 1);
```

Comment:
Why do 1 1 1 if we can pull out the location out of Pexp_construct?

Reply:
This is because the actual location of the boolean expression start is in the containing if _ then true else false expression, not in the Pexp_construct variant. I haven't developed a neat way of passing that information down effectively, so until then I'm leaving it as 1 1 1 for now so that dev work can proceed.

Reply:
Sounds good.